### PR TITLE
Removed Replication 2.

### PR DESCRIPTION
**Pull request by maierlars**
Remove Replication2 from 3.10.

### CODE REVIEW

File: arangod/Transaction/Methods.cpp [hunk 1]
```diff
@@ -1676,9 +1665,7 @@ Future<OperationResult> transaction::Methods::modifyLocal(
                           _state->isDBServer());
 
   bool excludeAllFromReplication =
-      replicationType != ReplicationType::LEADER ||
-      (followers->empty() &&
-       collection->replicationVersion() != replication::Version::TWO);
+      replicationType != ReplicationType::LEADER || (followers->empty());
```
Comment (goedderz) [inline arangod/Transaction/Methods.cpp:1668]:
```suggestion
      replicationType != ReplicationType::LEADER || followers->empty();
```

File: arangod/Transaction/Methods.cpp [hunk 2]
```diff
@@ -1825,9 +1812,7 @@ Future<OperationResult> transaction::Methods::modifyLocal(
 
   auto resDocs = resultBuilder.steal();
   if (res.ok()) {
-    if (replicationType == ReplicationType::LEADER &&
-        (!followers->empty() ||
-         collection->replicationVersion() == replication::Version::TWO) &&
+    if (replicationType == ReplicationType::LEADER && (!followers->empty()) &&
```
Comment (goedderz) [inline arangod/Transaction/Methods.cpp:1815]:
```suggestion
    if (replicationType == ReplicationType::LEADER && !followers->empty() &&
```

File: arangod/Transaction/Methods.cpp [hunk 3]
```diff
@@ -2028,9 +2013,7 @@ Future<OperationResult> transaction::Methods::removeLocal(
   }
 
   bool excludeAllFromReplication =
-      replicationType != ReplicationType::LEADER ||
-      (followers->empty() &&
-       collection->replicationVersion() != replication::Version::TWO);
+      replicationType != ReplicationType::LEADER || (followers->empty());
```
Comment (goedderz) [inline arangod/Transaction/Methods.cpp:2016]:
```suggestion
      replicationType != ReplicationType::LEADER || followers->empty();
```

File: arangod/Transaction/Methods.cpp [hunk 4]
```diff
@@ -2157,10 +2140,7 @@ Future<OperationResult> transaction::Methods::removeLocal(
 
   auto resDocs = resultBuilder.steal();
   if (res.ok()) {
-    auto replicationVersion = collection->replicationVersion();
-    if (replicationType == ReplicationType::LEADER &&
-        (!followers->empty() ||
-         replicationVersion == replication::Version::TWO) &&
+    if (replicationType == ReplicationType::LEADER && (!followers->empty()) &&
```
Comment (goedderz) [inline arangod/Transaction/Methods.cpp:2143]:
```suggestion
    if (replicationType == ReplicationType::LEADER && !followers->empty() &&
```

File: arangod/Cluster/ClusterInfo.cpp [hunk 5]
```diff
@@ -3557,13 +3308,8 @@ Result ClusterInfo::createCollectionsCoordinator(
...
 
-  auto replicatedStatesWait = std::invoke([&]() -> futures::Future<Result> {
-    if (replicationVersion == replication::Version::TWO) {
-      // TODO could the version of a replicated state change in the meantime?
-      return waitForReplicatedStatesCreation(databaseName, replicatedStates);
-    }
-    return Result{};
-  });
+  auto replicatedStatesWait =
+      std::invoke([&]() -> futures::Future<Result> { return Result{}; });
```
Comment (goedderz) [inline arangod/Cluster/ClusterInfo.cpp:3312]:
```suggestion
```

File: arangod/Cluster/Maintenance.cpp [hunk 6]
```diff
@@ -478,8 +468,7 @@ static void handleLocalShard(
    *      before it actually took ownership.
    */
 
-  if (replicationVersion != replication::Version::TWO &&
-      (activeResign || adjustResignState)) {
+  if ((activeResign || adjustResignState)) {
```
Comment (goedderz) [inline arangod/Cluster/Maintenance.cpp:471]:
```suggestion
  if (activeResign || adjustResignState) {
```

**Review by goedderz — APPROVED** [review]
LGTM. You can ignore the suggestions. I have not reviewed files that were deleted.

**Review by gnusi — APPROVED** [review]

**Pull request merged**